test/get_test.js

```diff
@@ -0,0 +1,34 @@
+goog.module('datalayerhelper.helper.testing.get');
+goog.setTestOnly();
+
+const {DataLayerHelper} = goog.require('helper');
+
+describe('The helper.get function', function() {
+  const h = new DataLayerHelper([{
+    a: 1,
+    b: {
+      c: {
+        d: 4,
+      },
+      e: 5,
+      f: null,
+    },
+  }]);
+
+  it('correctly returns the value currently assigned ' +
+      'to the given key in the helper internal model', function() {
+    expect(h.get('a')).toBe(1);
+    expect(h.get('b')).toEqual({c: {d: 4}, e: 5, f: null});
+    expect(h.get('b.c')).toEqual({d: 4});
+    expect(h.get('b.c.d')).toBe(4);
+    expect(h.get('b.e')).toBe(5);
+    expect(h.get('b.f')).toBe(null);
+  });
+
+  it('correct returns undefined when key does not exist in model', function() {
+    expect(h.get('blah')).toBe(undefined);
+    expect(h.get('c')).toBe(undefined);
+    expect(h.get('d')).toBe(undefined);
+    expect(h.get('e')).toBe(undefined);
```

Comment:
I realize this makes it not a direct port anymore since these aren't in the original test suite, but some additional tests that provide coverage for cases not already covered here:

- When a subfield is not defined but the parent is.
expect(h.get('b.blah')).toBe(undefined);
- When a subfield is requested but its parent isn't defined.
expect(h.get('b.blah.blah.blah')).toBe(undefined);

Reply:
Added, thanks the suggested cases!